account/identity.go

```diff
@@ -163,6 +164,45 @@ func (m *GormIdentityRepository) Create(ctx context.Context, model *Identity) er
 	return nil
 }
 
+// Lookup looks for an existing identity with the given `profileURL` or creates a new one
+func (m *GormIdentityRepository) Lookup(ctx context.Context, username, profileURL, providerType string) (*Identity, error) {
+	if username == "" || profileURL == "" {
```

Comment:
ha, sorry, I mixed `providerURL` with `providerType`. I could check that one, too, for sure, although I doubt it could be empty..